### DISCUSSION TITLE
Can't get type inference from createTRPCNext instance

### DISCUSSION THREAD

**Discussion by flythenimbus** [trpc/trpc#6255] [Q&A] [answered]
Hi everyone,

I have separate FE and BE using tRPC to communicate, in a monorepo environment. The FE app is a simple Next.js app, the BE is tRPC with Express. I'm using Bun as the runtime.

I'm exporting `AppRouter`  from the BE as per the instructions, and importing the type in the FE and passing it as a type argument to `createTRPCNext`. When inspecting `createTRPCNext` in the FE, I can see the type has resolved and the procedures are in there.

However, when I create the FE tRPC instance (`const trpc = createTRPCNext<AppRouter>({...})`), the `trpc` variable has a complex type attached to it (posted below), but nothing is really typed. when I do `trpc.example` (one of my procedures), it's `any`. When I try to get the tRPC HOC (`withTRPC`), that is `any` also. Why arent the types getting infered properly in the FE? That is what I'm trying to solve at the moment.

The type:

```
const trpc: ProtectedIntersection<CreateTRPCNextBase<TRouter, TSSRContext>, DecoratedProcedureRecord<TRouter["_def"]["record"], TFlags>>
```

Thanks!

**Comment by juliusmarminge** [comment]
Usually means one of your top-level routers/procedure has a name that conflicts with trpc internal properties (That's what the ProtectedIntersection is doing)

What top-level procedures/routers do you have?

**Reply by flythenimbus** [reply]
I have `userRouter` and `authRouter`. The procedure names: updateUser, getUsers, getUser, login, refreshToken, signup, logout, getAuth. All these are from a boilerplate project I found on [GitHub](https://github.com/alan345/TER).

**Reply by flythenimbus** [reply]
I wanted to go and check what `ProtectedIntersection` does, and to my surprise, it's showing this error:

```
Module '"@trpc/server"' has no exported member 'ProtectedIntersection'.
```

**Reply by juliusmarminge** [reply, marked as answer]
Are you Using the same version for your Client and server? It might be exported from a submodule in recent versions

**Reply by flythenimbus** [reply]
Turns out my `@trpc/server` was on @next, while `@trpc/next` was on 10.x. I just upgraded the next library to @next also. Thanks for the help @juliusmarminge !
